chain/trie/pruning.go

```diff
@@ -218,68 +218,79 @@ func (ref *RefCounts) Prune(hash common.Uint256, inMemory bool) error {
 	return nil
 }
 
-func (ref *RefCounts) prune(n node, inMemory bool) error {
+// refCountsIndex returns the RefCounts index for a given node
+// n.cache() is used for short/full nodes, hashNode itself for hashNode types
+func (ref *RefCounts) refCountsIndex(n node) (hs common.Uint256) {
+
+	var hash hashNode
 	switch n := n.(type) {
 	case *shortNode:
-		hash, _ := n.cache()
-		hs, _ := common.Uint256ParseFromBytes(hash)
-		count, ok := ref.counts[hs]
-		if !inMemory {
-			if !ok {
-				v, err := ref.trie.db.Get(db.TrieRefCountKey(hash))
-				if err == nil {
-					count = binary.LittleEndian.Uint32(v)
-					ref.counts[hs] = count
-				} else {
-					log.Fatal("Trie get error: %v", err)
-				}
+	case *fullNode:
+		hash, _ = n.cache()
+	case hashNode:
+		hash = n
+	default:
+		log.Fatalf("Invalid node type : %v, %v", reflect.TypeOf(n), n)
+	}
+	hs, _ = common.Uint256ParseFromBytes(hash)
```

Comment:
Error is still not handled

Reply:
And will not, I'm not a student of your's, stop this. https://github.com/nknorg/nkn/issues/924